lms/djangoapps/courseware/access.py

```diff
@@ -279,9 +279,32 @@ def can_view_courseware_with_prerequisites():  # pylint: disable=invalid-name
         else:
             return True
 
+    def can_view_courseware_with_entrance_exam():  # pylint: disable=invalid-name
```

Comment:
I feel like this check could become part of the proposed new `lms\djangoapps\courseware\entrance_exams.py` module, which would help to simplify things a bit.